### PR TITLE
feat(auth): `useIsCurrentUserLoaded()` composable

### PR DESCRIPTION
**Pull request by Soviut**
Closes https://github.com/vuejs/vuefire/issues/1306

### CODE REVIEW

File: src/auth/user.ts [hunk 1]
```diff
@@ -36,6 +36,16 @@ export function useCurrentUser(name?: string) {
...
 }
 
+/**
+ * Returns a boolean indicating if the current user has finished loading and is no longer undefined. This is useful for
+ * hiding navigation until the current user is loaded and we know if they are authenticated or not.
+ * @param name - name of the application
+ */
+export function isUserLoaded(name?: string) {
+  const currentUser = useCurrentUser(name)
+  return computed(() => typeof currentUser !== 'undefined')
```
Comment (posva) [inline src/auth/user.ts:46]:
```suggestion
  return computed(() => currentUser !== undefined)
```

File: src/auth/user.ts [hunk 2]
```diff
@@ -36,6 +36,16 @@ export function useCurrentUser(name?: string) {
   return authUserMap.get(useFirebaseApp(name))!
 }
 
+/**
+ * Returns a boolean indicating if the current user has finished loading and is no longer undefined. This is useful for
+ * hiding navigation until the current user is loaded and we know if they are authenticated or not.
+ * @param name - name of the application
+ */
+export function isUserLoaded(name?: string) {
```
Comment (posva) [inline src/auth/user.ts:44]: Let's name this `useIsUserLoaded()`. The `use` prefix is important in this case

File: src/auth/user.ts [hunk 3]
```diff
@@ -36,6 +36,16 @@ export function useCurrentUser(name?: string) {
   return authUserMap.get(useFirebaseApp(name))!
 }
 
+/**
+ * Returns a boolean indicating if the current user has finished loading and is no longer undefined. This is useful for
+ * hiding navigation until the current user is loaded and we know if they are authenticated or not.
+ * @param name - name of the application
```
Comment (posva) [inline src/auth/user.ts:42]:
```suggestion
 *
 * @param name - name of the application
```

File: src/auth/user.ts [hunk 4]
```diff
@@ -36,6 +36,16 @@ export function useCurrentUser(name?: string) {
   return authUserMap.get(useFirebaseApp(name))!
 }
 
+/**
+ * Returns a boolean indicating if the current user has finished loading and is no longer undefined. This is useful for
+ * hiding navigation until the current user is loaded and we know if they are authenticated or not.
```
Comment (posva) [inline src/auth/user.ts:41]: Let's use this and rewrap around lines:


```suggestion
 * Helper that returns a computed boolean that becomes `true` as soon as the current user is no longer `undefined`. Note this doesn't ensure the user is logged in, only if the initial signing process has run.
```

**Review by posva — CHANGES_REQUESTED** [review]
Thanks!

File: src/auth/user.ts (hunk 2 above)
Comment (Soviut) [reply src/auth/user.ts:44]: You're right, it was late and I wasn't thinking. Of course it's a composable.

**Comment by Soviut** [comment]
@posva I also added docs. Let me know if those are okay.

**Review by posva — APPROVED** [review]

**Pull request merged**